Python/pystate.c

```diff
@@ -949,6 +1110,14 @@ tstate_delete_common(PyThreadState *tstate,
     }
 }
 
+static void
+free_threadstate(PyThreadState *tstate)
+{
+    if (tstate != &tstate->interp->_preallocated.tstate) {
```

Comment:
Why is the allocation status of a thread stored in the interpreter state?
How can that work?

Reply:
You mean the "preallocated" thread state noted here?  I'm not sure what you're asking.  In this function we free `tstate` only if it was *not* statically allocated.

Reply:
Why not store the allocation status on the thread?
`if (tstate != &tstate->interp->_preallocated.tstate)` then becomes `if (tstate->allocated)`

Reply:
I'm dropping "initialized".  It isn't needed any more since the simplification I did earlier.

Reply:
Good point.

Reply:
I fixed this.